packages/ember-glimmer/lib/utils/process-args.js

```diff
@@ -1,35 +1,45 @@
+import { assert } from 'ember-debug';
 import { symbol } from 'ember-utils';
 import { ARGS } from '../component';
 import { UPDATE } from './references';
+import { DEBUG } from 'ember-env-flags';
 import { MUTABLE_CELL } from 'ember-views';
 import { ACTION } from '../helpers/action';
 
-// ComponentArgs takes EvaluatedNamedArgs and converts them into the
+// processComponentArgs takes CapturedNamedArguments and converts them into the
 // inputs needed by CurlyComponents (attrs and props, with mutable
 // cells, etc).
-export function processComponentArgs(namedArgs) {
+export function processComponentArgs(namedArgs, componentName) {
   let keys = namedArgs.names;
   let attrs = namedArgs.value();
   let props = Object.create(null);
   let args = Object.create(null);
 
-  props[ARGS] = args;
+  let seen;
+  if (DEBUG) {
+    seen = Object.create(null);
+  }
 
   for (let i = 0; i < keys.length; i++) {
     let name = keys[i];
+
+    assert(
+      `Component attribute \`${name}\` is declared multiple times for \`${componentName.split(':').slice(-1)}\` component.`,
+      seen[name] ? false : seen[name] = true
+    );
```

Comment:
Is there any reason this check cannot be done in one pass in L20?

Something like:

```js
if (DEBUG) {
  let repeatedKeys = Array.from(keys).filter((key, idx, coll) => coll.indexOf(key) !== idx);
  assert(
    `Component attributes ${repeatedKeys.join(', ')} are declared multiple times for \`${componentName.split(':').slice(-1)}\` component.`,
    repeatedKeys.length === 0
  )
}
```

Reply:
might not work `Array.from([1,2,3,4,4,4,4]).filter((key, idx, coll) => coll.indexOf(key) !== idx)`
will give `[4, 4, 4]`
```

  if (DEBUG) {
    let repeatedKeys = keys.reduce(function(sum, key) {
      sum[key] = 0 || sum[key];
      sum[key]++;
      return sum;
    }, {});

    repeatedKeys = Object.keys(repeatedKeys).filter((key)=> repeatedKeys[key] > 1);

    assert(
      `Component attribute \`${repeatedKeys.join(', ')}\` is declared multiple times for \`${componentName.split(':').slice(-1)}\` component.`,
      repeatedKeys.length === 0
    );
  }
```
is this better ?

Reply:
I like how this reads, much clearer to understand.